linkerd/app/inbound/src/lib.rs

```diff
@@ -159,13 +168,20 @@ impl<A: OrigDstAddr> Config<A> {
                 .push(normalize_uri::layer())
                 .push(http_target_observability)
                 .into_new_service()
-                .push_on_response(
-                    svc::layers()
-                        .push_lock()
-                        .push(metrics.stack.layer(stack_labels("target"))),
+                .cache(
+                    svc::layers().push_on_response(
+                        svc::layers()
+                            // If the service has been ready & unused for `cache_max_idle_age`,
+                            // fail it.
+                            .push_idle_timeout(cache_max_idle_age)
+                            // If the service has been unavailable for an extend time, eagerly
+                            // fail requests.
+                            .push_failfast(Duration::from_secs(10))
+                            // Shares the service, ensuring discovery errors are propagated.
+                            .push_spawn_buffer(10)
```

Comment:
seems like this shows up enough that we might want to have a `push_idle_timeout_failfast_and_buffer` or something (but please don't call it that!)

Reply:
to wit, if  the _only_ time we push these layers is all together, it doesn't really seem like they all need their own `push_` methods on `Stack`, but it _does_ seem like we might want a `Stack::push_all_these_layers_together` method...

Reply:
I'd rather not for now, as this makes it all more brittle to change. If it stays this way for a while, we can formalize it.

Reply:
fair enough. there is definitely a  tradeoff between, on one hand, wanting to be able to change every instance of this in one place, and on the other hand, not wanting that. :)